api/routers/feature.py

```diff
@@ -0,0 +1,140 @@
+    ] = None,
+    f: Annotated[
+        formatters.Metadata_Formats, Query(description="Specify return format")
+    ] = formatters.Metadata_Formats.geojson,
+):
+    if not bbox and not platform:
+        raise HTTPException(400, detail="Have to set at least one of bbox or platform.")
+    if bbox:
+        left, bottom, right, top = map(str.strip, bbox.split(","))
+        poly = geometry.Polygon([(left, bottom), (right, bottom), (right, top), (left, top)])
+    if datetime:
+        range = get_datetime_range(datetime)
+
+    obs_request = dstore.GetObsRequest(
+        filter=dict(
+            metadata_id=dstore.Strings(values=ids.split(",") if ids else None),
+            parameter_name=dstore.Strings(values=parameter_name.split(",") if parameter_name else None),
+            naming_authority=dstore.Strings(values=naming_authority.split(",") if naming_authority else None),
+            institution=dstore.Strings(values=institution.split(",") if institution else None),
+            platform=dstore.Strings(values=platform.split(",") if platform else None),
+            standard_name=dstore.Strings(values=standard_name.split(",") if standard_name else None),
+            unit=dstore.Strings(values=unit.split(",") if unit else None),
+            instrument=dstore.Strings(values=instrument.split(",") if instrument else None),
+            level=dstore.Strings(values=level.split(",") if level else None),
+            period=dstore.Strings(values=period.split(",") if period else None),
+            function=dstore.Strings(values=function.split(",") if function else None),
+        ),
+        spatial_area=(
+            dstore.Polygon(points=[dstore.Point(lat=coord[1], lon=coord[0]) for coord in poly.exterior.coords])
+            if bbox
+            else None
+        ),
+        temporal_interval=(dstore.TimeInterval(start=range[0], end=range[1]) if datetime else None),
+        temporal_mode="latest",
+    )
+
+    time_series = await get_obs_request(obs_request)
+
+    return formatters.metadata_formatters[f](time_series)
+
+
+@router.get("/items/{item_id}", tags=["Collection items"], response_model=Feature, response_model_exclude_none=True)
+async def get_time_series_by_id(
+    item_id: Annotated[str, Path()],
+    f: Annotated[
+        formatters.Metadata_Formats, Query(description="Specify return format")
+    ] = formatters.Metadata_Formats.geojson,
+):
+    obs_request = dstore.GetObsRequest(filter=dict(metadata_id=dstore.Strings(values=[item_id])))
+    time_series = await get_obs_request(obs_request)
+
+    return formatters.metadata_formatters[f](time_series)
+
+
+@router.get("/dataset", tags=["E-SOH dataset"], include_in_schema=False)
+async def get_dataset_metadata():
+    # need to get spatial extent.
+    spatial_request = dstore.GetExtentsRequest()
+    extent = await get_spatial_extent(spatial_request)
+    dynamic_fields = {
+        "spatial_extents": [
```

Comment:
I think a GeoJSON polygon has to be closed:
https://datatracker.ietf.org/doc/html/rfc7946#section-3.1.6

I think there also needs to be an extra layer of square brackets:
https://datatracker.ietf.org/doc/html/rfc7946#appendix-A.3